src/main/java/server/managers/BookingManager.java

```diff
@@ -61,67 +60,275 @@ public static LocalTime getCloseTime() {
             }
 
             // if the last booking ends at closing time
-            if(availStart.toLocalTime().compareTo(closeTime) == 0){
+            if(availStart.compareTo(closeTime) == 0){
                 return new Pair<ArrayList, Exception>( results, null);
             }
-            slot = new ArrayList<LocalDateTime>();
+            slot = new ArrayList<LocalTime>();
             slot.add(availStart);
-            slot.add(closeTime.atDate(currDate));
+            slot.add(closeTime);
             results.add(slot);
         }
 
         return new Pair<ArrayList, Exception>( results, null);
     }
 
-    public void sortBookings(ArrayList<Booking> bookingsList){
-        Collections.sort(bookingsList, Booking.BookingComparator);
+
+
+    public Pair<Booking, Exception> createBooking(String day, int bookingId, int clientId, String facilName, LocalTime newStart, LocalTime newEnd, Hashtable facilTable){
+
+        Exception e;
+
+        e = doBookingCheck(facilName,newStart,newEnd,facilTable);
+        if (e != null){
+            return new Pair<Booking, Exception>(null, e);
+        }
+        e = doAvailabilityCheck(day, facilName,newStart,newEnd,facilTable);
+        if (e != null){
+            return new Pair<Booking, Exception>(null, e);
+        }
+
+        //Booking procedure
+        Booking b = new Booking(day, bookingId, clientId, facilName, newStart, newEnd);
+        Facility f = (Facility) facilTable.get(facilName);
+        f.addBooking(b);
+        return new Pair<Booking, Exception>(b, null);
     }
 
-    public Pair<Booking, Exception> createBooking(int bookingId, int clientId, String facilName, LocalDateTime newStart, LocalDateTime newEnd, Hashtable facilTable){
+    public Pair<Booking, Exception> offsetBooking(String facilName, int bookingId, int offset, Hashtable facilTable){
+        //FIXME offset should be in minutes
+        Exception e;
 
-        facilName = facilName.toLowerCase();
+        Booking b = this.findFacilBookingById(facilName, bookingId, facilTable);
+        String day = b.getDay();
 
-        if (!facilTable.containsKey(facilName)){
-            Exception e = new NoSuchElementException("Facility does not exist"); //TODO review this exception
+        LocalTime newStart = b.getStart().plusMinutes(offset);
+        LocalTime newEnd = b.getEnd().plusMinutes(offset);
```

Comment:
in the client side, i designed offset to be in slots instead of minutes. but either way works.

Reply:
yeap i noticed, this is one of the things i will coordinate w xq when we integrating our parts